File: src/main/java/io/github/kingg22/api/vacunas/panama/persistence/repository/ProvinciaRepository.java
package io.github.kingg22.api.vacunas.panama.persistence.repository;

import io.github.kingg22.api.vacunas.panama.persistence.entity.Provincia;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface ProvinciaRepository extends JpaRepository<Provincia, Short> {

    Optional<Provincia> findByNombre(String nombre);
}

File: src/main/kotlin/io/github/kingg22/api/vacunas/panama/modules/direccion/repository/ProvinciaRepository.kt
package io.github.kingg22.api.vacunas.panama.modules.direccion.repository

import io.github.kingg22.api.vacunas.panama.modules.direccion.entity.Provincia
import io.github.kingg22.api.vacunas.panama.util.withSession
import io.quarkus.hibernate.reactive.panache.kotlin.PanacheRepositoryBase
import io.smallrye.mutiny.coroutines.awaitSuspending
import jakarta.enterprise.context.ApplicationScoped

@ApplicationScoped
class ProvinciaRepository : PanacheRepositoryBase<Provincia, Short> {
    suspend fun listAllProvincias() = withSession { listAll().awaitSuspending() }
}
